### ISSUE TITLE
unwrap-used not triggering for DashMap

### ISSUE THREAD

**Issue by laralove143** [rust-lang/rust-clippy#8399]
### Summary

this code works as expected
```rs
let mut hm = HashMap::new();
hm.insert("foo", "bar");
hm.get("foo").unwrap();
```
but
```rs
let dm = DashMap::new();
dm.insert("foo", "bar");
dm.get("foo").unwrap();
```
does not trigger any lint

### Lint Name

unwrap-used

### Reproducer

I tried this code:

```rust
let dm = DashMap::new();
dm.insert("foo", "bar");
dm.get("foo").unwrap();
```

I expected to see this happen:
`clippy::unwrap-used` triggers
Instead, this happened:
nothing

### Version

```text
rustc 1.58.1 (db9d1b20b 2022-01-20)
binary: rustc
commit-hash: db9d1b20bba1968c1ec1fc49616d4742c1725b4b
commit-date: 2022-01-20
host: aarch64-apple-darwin
release: 1.58.1
LLVM version: 13.0.0
```

**Comment by giraffate** [comment]
The `unwrap_used` didn't trigger for `get()` or `get_mut()`, but it will be fixed in https://github.com/rust-lang/rust-clippy/issues/8124.

**Comment by laralove143** [comment]
that seems different, like i said in the comment it works as expected with vecs and hashmaps

**Comment by giraffate** [comment]
Hm, at least for now, clippy doesn't seem to trigger the `unwrap_used` lint for `HashMap`. Do I get that right?
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=5ba51b74b72a57a8a6150538936fe690

**Comment by laralove143** [comment]
yeah but it triggers get_unwrap so it’s fine, except DashMap doesn’t implement Index so with any other similar case it’d be a problem. i wonder why this acts different than any other unwraps

**Comment by laralove143** [comment]
is there any updates on this?

**Comment by sgued** [comment]
Should this still be open? I can't reproduce it with any clippy version > 1.60
